app/services/xccdf_report_parser.rb

```diff
@@ -51,21 +58,65 @@ def check_for_external_reports
     # rubocop:disable Style/GuardClause
     if external_report?
       raise ExternalReportError,
-            "No policy found matching benchmark #{benchmark.ref_id} "\
-            "(RHEL-#{benchmark.os_major_version}) and profile "\
-            "#{test_result_profile.ref_id} with host #{@host.name} assigned "\
-            "for account #{@account.account_number}."
+            "No policy found matching benchmark #{benchmark.ref_id}. "\
+            "#{parse_failure_message}"
     end
     # rubocop:enable Style/GuardClause
   end
 
+  def check_for_missing_benchmark
+    # rubocop:disable Style/GuardClause
+    unless benchmark.persisted?
+      raise UnknownBenchmarkError,
+            "No benchmark found matching ref_id #{benchmark.ref_id} and "\
+            "SSG #{benchmark.version}. #{parse_failure_message}"
+    end
+    # rubocop:enable Style/GuardClause
+  end
+
+  def check_for_missing_test_result_profile
+    # rubocop:disable Style/GuardClause
+    unless test_result_profile.persisted?
+      raise UnknownProfileError,
+            "No profile found matching ref_id #{test_result_profile.ref_id} "\
+            "in benchmark #{benchmark.ref_id} with SSG "\
+            "#{benchmark.version}. #{parse_failure_message}"
+    end
+    # rubocop:enable Style/GuardClause
+  end
+
+  def check_for_missing_rules
+    # rubocop:disable Style/GuardClause
+    if test_result_rules_missing.any?
```

Comment:
Can you please elaborate on this check?

If this checks whether the test result contains more rules that there are in the stored benchmark, then this should not fail but rather set is as unsupported or skip the check at all. It this should somehow compare rule sets of between test result and stored benchmark it would likely need a different approach.

Reply:
See https://github.com/RedHatInsights/compliance-backend/pull/681#discussion_r542400864